Fetch only the commit we want from each dependency's repository.
--- skia/tools/git-sync-deps.orig	2023-04-19 21:10:08.000000000 -0500
+++ skia/tools/git-sync-deps	2023-04-19 21:17:19.000000000 -0500
@@ -131,9 +131,12 @@
   Raises an exception if any calls to git fail.
   """
   if not os.path.isdir(directory):
+    os.makedirs(directory)
     subprocess.check_call(
-      [git, 'clone', '--quiet', '--no-checkout', repo, directory])
-    subprocess.check_call([git, 'checkout', '--quiet', commithash],
+      [git, '-c', 'init.defaultBranch=main', 'init', '-q'], cwd=directory)
+    subprocess.check_call([git, 'remote', 'add', 'origin', repo], cwd=directory)
+    subprocess.check_call([git, 'fetch', '--depth', '1', 'origin', commithash], cwd=directory)
+    subprocess.check_call([git, '-c', 'advice.detachedHead=false', 'checkout', '-q', 'FETCH_HEAD'],
                           cwd=directory)
     if verbose:
       status(directory, commithash, True)
